### PR TITLE
Removed the product check for client API < 8

### PR DESCRIPTION
**Pull request by ezimuel**
This PR should solve #227 removing the `x-elastic-product` check for API version < 8.

### CODE REVIEW

File: lib/Search/Elasticsearch.pm [hunk 1]
```diff
@@ -64,6 +65,9 @@ sub new {
         $plugin_class->_init_plugin($params);
     }
 
+    # Get major version of client API
+    $Search::Elasticsearch::API_VERSION = substr($params->{client}->{api_version}, 0, index($params->{client}->{api_version}, '_'));
```
[the remark this answers is not in the document]
Comment (ezimuel) [reply lib/Search/Elasticsearch.pm:69]: Good catch, even if this apply only if you have multiple instances of the client.  I need to find a way to access the API version of the client from `process_response()` in Search::Elasticsearch::Role::Cxn. I'm looking into it.

**Comment by ezimuel** [comment]
@Grinnz Apart from the improvement that you suggested, did you try the fix? Thanks.

**Comment by stuartskelton** [comment]
@ezimuel  I have tested this and it works.  I currently only work with multiple ES clusters with the same version (7.10.0) so the Global does  not affect me.

I have rolled back to v7.3 so I can avoid this issue.

@Grinnz do you have any other feed back?

**Comment by ezimuel** [comment]
@stuartskelton thanks for the feedback. I will try merge this asap and have a release in place.

**Comment by stuartskelton** [comment]
same here it looks reasonable to me.

**Comment by Grinnz** [comment]
I've tested Search-Elasticsearch-8.00 with this patch with elasticsearch server 6.8.23 and everything looks good.

**Pull request merged**